TumbleBitSetup/PoupardStern/Implementation/PoupardStern.cs

```diff
@@ -172,19 +191,19 @@ public static bool Verifying(RsaPubKey pubKey, BigInteger[] xValues, BigInteger
         /// <param name="k">Security parameter specified in the setup.</param>
         /// <param name="keyLength">The size of the RSA key in bits</param>
         /// <returns></returns>
-        internal static BigInteger SampleFromZnStar(RsaPubKey pubKey, byte[] psBytes, int i, int BigK, int keyLength)
+        internal static BigInteger SampleFromZnStar(RsaKeyParameters pubKey, byte[] psBytes, int i, int BigK, int keyLength)
         {
-            BigInteger Modulus = pubKey._pubKey.Modulus;
+            BigInteger Modulus = pubKey.Modulus;
 
             // Octet Length of i
             int iLen = Utils.GetOctetLen(BigK);
             // OctetString of i
             var EI = Utils.I2OSP(i, iLen);
             // ASN.1 encoding of the PublicKey
-            var keyBytes = pubKey.ToBytes();
+            var keyBytes = ToBytes(pubKey);
```

Comment:
This way of casting public keys to Bytes is not consistent with the way it's done in PermutationTest.cs ([here](https://github.com/osagga/TumbleBitSetup/pull/23/commits/e070730a6e1dd45fef5e37f5df52148e7cb2f959#diff-91dbf9ad7f5b8229a9b159fc6d6e24a4R186)) given that you added ToBytes to the Extensions.cs